scripts/fastlane/Fastfile

```diff
@@ -0,0 +1,192 @@
+    # Default to mode = 1 (derive from new version name)
+    mode = options[:mode] != nil ? options[:mode] : 1
+    segments = options[:segments]
+    prerelease = segments.length() > 3 ?  0 : segments[-1].to_i
+    # Android version code max version = 2_100_000_000
+
+    case mode
+    when 1
+      # ======= option 1 =======
+      # Derive the version code from version segements and pre-release number (if applicable)
+      # With this approach, you're "limited" to 999 minor versions per major and 999 patches per minor version.
+      # Limits: Max version for 1.x would be 1.999.999 with version code 199_999_900
+      version_code = segments[0].to_i * 100_000_000 + segments[1].to_i * 100_000 + segments[2].to_i * 100 + prerelease
+    when 2
+      # ======= option 2 =======
+      # Date-based approach 1
+      # We can only do <YEAR><MONTH><DAY><HR> because we'd be over the limit if we were to include minutes
+      # Reference: https://www.rockandnull.com/update-app-version-android/
+      # Limits: one release per hour max.=> Result would look like 21031002
+      version_code = Time.now.utc.strftime("%y%m%d%H").to_i 
+    when 3
+      # ======= option 3 =======
+      # Or include minute and subtract 1/1/2021 00:00 from Time.now.utc.strftime("%y%m%d%H%M").to_i
+      # Limits: one release per minute => Result would look like 2_090_229
+      version_code = Time.now.utc.strftime("%y%m%d%H%M").to_i - 2_101_010_000 
+    end
+    version_code
+  end
+
+  desc "Create the release commit."
+  private_lane :release_commit do |options|
+    next_version = options[:version]
+    commit_message = "chore: release #{next_version} [skip ci]"
+    sh('git', 'commit', '-am', commit_message)
+  end
+
+  desc "Tag in git and push to GitHub"
+  private_lane :add_tag do |options|
+    next_version = options[:version]
+
+    add_git_tag(tag: next_version)
+    push_git_tags(tag: next_version)
+  end
+
+  desc "Publish release to maven using attributes from environment variables."
+  private_lane :publish_to_maven do |options|
+    gradle(
+      project_dir: gradle_project_root,
+      task: "uploadArchives",
+      print_command: false,
+      flags: '--stacktrace',
+      properties: {
+        "SONATYPE_NEXUS_USERNAME": ConfigValueReader.from_environment_variable('SONATYPE_NEXUS_USERNAME'),
+        "SONATYPE_NEXUS_PASSWORD": ConfigValueReader.from_environment_variable('SONATYPE_NEXUS_PASSWORD'),
+        "signing.inMemoryKey": ConfigValueReader.from_environment_variable('signing.inMemoryKey'),
+        "signing.password": ConfigValueReader.from_environment_variable('signing.password'),
+        "signing.keyId": ConfigValueReader.from_environment_variable('signing.keyId'),
+        "signing.secretKeyRingFile": ConfigValueReader.from_environment_variable('signing.secretKeyRingFile')
+      }
+    )
+  end
+
+  desc "Publish release to maven without overriding gradle properties."
+  private_lane :publish_to_maven_no_overrides do |options|
```

Comment:
We probably don't need this one, right?  We can just use the approach above, with environment variables?

Reply:
Yeah...I can remove this one. Could be useful if we were to put the credentials in the gradle home directory, but we're unlikely to do that in a CI scenario.